File: TPS/TP1/Recusivos/Parecursivo.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdbool.h>

int verifica() {
    char texto[1000];

    fgets(texto, sizeof(texto), stdin);
    texto[strcspn(texto, "\n")] = '\0';

    if (strcmp(texto, "FIM") == 0) {
        return 2;
    }

    return ehPalindromo(texto, 0);
}

int ehPalindromo(char texto[], int i) {
    int tam = strlen(texto);

    if (i >= tam / 2) {
        return 1;
    }

    if (texto[i] != texto[tam - i - 1]) {
        return 0;
    }

    return ehPalindromo(texto, i + 1);
}

int main() {   
    int teste;

    do {
        teste = verifica();  
        if (teste == 1) {
            printf("SIM\n");
        } else if (teste == 0) {
            printf("NAO\n");
        }
    } while (teste != 2);

    return 0;
}

File: TPS/TP1/Recusivos/Parecursivo.java
//package TP1.Recusivos;
import java.util.*;

public class Parecursivo {

    public static boolean verifica(String texto) {
        if ((texto.charAt(0) == 'F' && texto.charAt(1) == 'I') && (texto.charAt(2) == 'M')) {
            return false;
        }
        return true;
    }

    public static boolean ehPalindromo(String texto, int i) {
        int tam = texto.length();
        
       
        if (i >= tam / 2) {
            return true;
        }

       
        if (texto.charAt(i) != texto.charAt(tam - i - 1)) {
            return false;
        }

        
        return ehPalindromo(texto, i + 1);
    }

    public static void main(String[] args) {
        Scanner ler = new Scanner(System.in);
        String texto = ler.nextLine();
        
      
        while (verifica(texto)) {
            
            if (ehPalindromo(texto, 0)) {
                System.out.println("SIM");
            } else {
                System.out.println("NAO");
            }

            texto = ler.nextLine();
        }

        ler.close();
    }
}